src/Controller/Order.php

```diff
@@ -3,35 +3,67 @@
 namespace App\Controller;
 
 use App\FakeDB;
+use App\LegacyDB;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 
+//use App\Entity\Order;
+
+
 final class Order
 {
     public function getAll()
     {
         //todo
         $orders = FakeDB::getOrders();
-        return new JsonResponse();
+      //  $ArrayOrders = new array();         //je laisse la preuve de mes essais en commentaire
+        foreach ($orders as $order) {   // je fait un foreach pour pouvoir récuperer des objets unique et appeller des méthodes sur chacun d'entre eux
```

Comment:
Ta solution est parfaite 👍 Il existe une manière de faire sans utiliser `for/foreach` une idée ?

Reply:
avec la fonction php : array_map() ?

Reply:
En passant la méthode toArray en static de la classe Entity/Order.php, on peut utiliser la méthode php : array_map, comme cela :  
`  $ArrayOrders = array_map(Order::toArray(),$orders); `
mais le contrôleur Order.php et la classe Order.php on le même nom du coup cela ne marche pas a la compilation.